packages/core/src/payment/payment-request-transformer.ts

```diff
@@ -166,24 +168,54 @@ export default class PaymentRequestTransformer {
         values: HostedInputValues,
         payment: HostedCreditCardInstrument | HostedVaultedInstrument,
         nonce: string,
+        paymentMethod?: PaymentMethod,
     ): PaymentInstrument {
-        return 'instrumentId' in payment
-            ? {
-                  ...payment,
-                  ccCvv: values.cardCodeVerification,
-                  ccNumber:
-                      values.cardNumberVerification &&
-                      this._cardNumberFormatter.unformat(values.cardNumberVerification),
-                  hostedFormNonce: nonce,
-              }
-            : {
-                  ...payment,
-                  ccCvv: values.cardCode,
-                  ccExpiry: this._cardExpiryFormatter.toObject(values.cardExpiry || ''),
-                  ccName: values.cardName || '',
-                  ccNumber: this._cardNumberFormatter.unformat(values.cardNumber || ''),
-                  hostedFormNonce: nonce,
-              };
+        const result =
+            'instrumentId' in payment
+                ? {
+                      ...payment,
+                      ccCvv: values.cardCodeVerification,
+                      ccNumber:
+                          values.cardNumberVerification &&
+                          this._cardNumberFormatter.unformat(values.cardNumberVerification),
+                      hostedFormNonce: nonce,
+                  }
+                : {
+                      ...payment,
+                      ccCvv: values.cardCode,
+                      ccExpiry: this._cardExpiryFormatter.toObject(values.cardExpiry || ''),
+                      ccName: values.cardName || '',
+                      ccNumber: this._cardNumberFormatter.unformat(values.cardNumber || ''),
+                      hostedFormNonce: nonce,
+                  };
+
+        if (
+            paymentMethod?.gateway === 'bluesnapdirect' &&
+            paymentMethod?.id === 'credit_card' &&
+            isHybridHostedFormInstrumentLike(result)
+        ) {
+            return this._transformToBlueSnapDirectCreditCardInstrument(result);
+        }
+
+        return result;
+    }
+
+    private _transformToBlueSnapDirectCreditCardInstrument({
+        nonce: pfToken,
+        hostedFormNonce,
+        ccName: cardHolderName,
+    }: HybridHostedFormInstrument): PaymentInstrument {
+        return {
+            formattedPayload: {
+                credit_card_token: {
+                    token: JSON.stringify({
+                        pfToken,
+                        cardHolderName,
+                    }),
+                },
+                hostedFormNonce,
+            },
+        };
```

Comment:
Similar to prior comments. How about putting this inside the Bluesnap package and using the payment integration API's `submitPayment(payment: Payment): Promise<PaymentIntegrationSelectors>` to submit Bluesnap required data?

Reply:
AFAIK it's not possible to retrieve the `cardHolderName` on the BlueSnap package because this value lives on another iframe. If I could access this value, the payment integration API's `submitPayment` method would be sufficient. Although I don't think it's a good idea.

Reply:
Undone.